packages/cli/src/commands/devHandler.js

```diff
@@ -23,6 +24,71 @@ export const handler = async ({
 }) => {
   const rwjsPaths = getPaths()
 
+  // Starting values of ports from config (redwood.toml)
+  let apiPreferredPort = parseInt(getConfig().api.port)
+  let webPreferredPort = parseInt(getConfig().web.port)
+
+  // Assume we can have the ports we want
+  let apiAvailablePort = apiPreferredPort
+  let apiPortChangeNeeded = false
+  let webAvailablePort = webPreferredPort
+  let webPortChangeNeeded = false
+
+  // Check api port
+  if (side.includes('api')) {
+    apiAvailablePort = await getFreePort(apiPreferredPort)
+    if (apiAvailablePort === -1) {
+      errorTelemetry(
+        process.argv,
+        `Error could not determine a free port for the api server`
+      )
+      console.error(`Error could not determine a free port for the api server`)
+      process.exit(-1)
```

Comment:
(See another similar suggestion)

```suggestion
      process.exit(1)
```